src/lib/Transform.ts

```diff
@@ -229,11 +236,27 @@ export function batch<T>(
 				flushError = e;
 			}
 
-			await writable.end(error, readable.result().catch(noop));
-			throwIfThrowable(earlyFlushError);
-			throwIfThrowable(flushError);
+			const toThrow = earlyFlushError || flushError;
+
+			await writable.end(
+				error,
+				toThrow
+					? readable.result().catch((e) => {
+							if (e !== toThrow) {
+								throw e;
+							}
+					  })
+					: readable.result()
+			);
+
+			throwIfThrowable(toThrow);
 		},
-		flush
+		() => {
+			if (!pendingWrite) {
```

Comment:
Seems odd to explicitly go and write stuff, when the idea of abort is to *not* write stuff anymore.
If the intention is to bounce the abort error back, it's probably cleaner to that instead?

Reply:
There are parallel requirements here with `write()/end()`:
1. If nothing is in the queue, don't try to write (i.e. do not throw `abortReason`)
2. If an error is thrown (in this case *when* the error is thrown), try to handle it with `handleError` optional parameter first
3. Every item read by `batch` should be written downstream, even if just to throw an (in this case predictable) error.

It feels "clean" to me that this can just hook on to the logic of the other cases, even if the outcome is obvious.

Reply:
Hmyeah, I forgot about `handleError`.

But in that case, suppose that `minBatchSize` is 10, and there are 3 remaining items in the queue while a previous `pendingWrite` is in progress. Will these remaining three be handled with this call? Or perhaps they other way around, if a stream is aborted, it still *also* needs to still be ended. Won't the ender already take care of this then?

Reply:
If there are 3 remaining items in the queue, `earlyFlush` will loop again, so it will be handled there instead of here.

In truth, we can get rid of this completely because `end()` takes care of all of it (just after a possible delay). If `abort()` is called after `end()` no action is required because `end()` will already have emptied the queue.

The only benefit is that the error will throw a little sooner. That may not be worth it for the extra complexity it introduces.

Reply:
Ah, yes, the error would be thrown, but only after `end()` has been called, which may be quite a bit after `abort()`.
I think I finally understand ;)

Reply:
Oh, and I think it's fine to leave it in, then.